sys/include/checksum/crc16_ccitt.h

```diff
@@ -37,8 +34,129 @@
+ * @param[in]  buf  Start of the memory area to checksum
+ * @param[in]  len  Number of bytes to checksum
+ *
+ * @return          Checksum of the specified memory area based on the
+ *                  given start value
+ */
+static inline uint16_t crc16_ccitt_mcrf4xx_update(uint16_t crc, const unsigned char *buf,
+                                                  size_t len)
+{
+    /* Since CCITT-KERMIT and CCITT-MCRF4XX only differ in the starting
+     * seed, we wrap around crc16_ccitt_kermit_update() for updating */
+    return crc16_ccitt_kermit_update(crc, buf, len);
+}
+
+/**
+ * @brief           Calculate CRC16-CCITT-MCRF4XX
+ * @details         polynom=0x1021
+ *                  seed=0xffff
+ *                  check=0x6f91
+ *
+ * @param[in]  buf  Start of the memory area to checksum
+ * @param[in]  len  Number of bytes to checksum
+ *
+ * @return          Checksum of the specified memory area
+ */
+uint16_t crc16_ccitt_mcrf4xx_calc(const unsigned char *buf, size_t len);
+
+/**
+ * @brief           Update CRC16-CCITT-FALSE
+ *
+ * @param[in]  crc  A start value for the CRC calculation, usually the
+ *                  return value of a previous call to
+ *                  crc16_ccitt_false_calc() or crc16_ccitt_false_update()
+ * @param[in]  buf  Start of the memory area to checksum
+ * @param[in]  len  Number of bytes to checksum
+ *
+ * @return          Checksum of the specified memory area based on the
+ *                  given start value
+ */
+uint16_t crc16_ccitt_false_update(uint16_t crc, const unsigned char *buf, size_t len);
+
+/**
+ * @brief           Calculate CRC16-CCITT-FALSE
+ * @details         polynom=0x1021
+ *                  seed=0xffff
+ *                  check=0x29b1
+ *
+ * @param[in]  buf  Start of the memory area to checksum
+ * @param[in]  len  Number of bytes to checksum
+ *
+ * @return          Checksum of the specified memory area
+ */
+uint16_t crc16_ccitt_false_calc(const unsigned char *buf, size_t len);
+
+/**
+ * @brief           Update CRC16-CCITT-AUG
+ *
+ * @param[in]  crc  A start value for the CRC calculation, usually the
+ *                  return value of a previous call to
+ *                  crc16_ccitt_aug_calc() or crc16_ccitt_aug_update()
+ * @param[in]  buf  Start of the memory area to checksum
+ * @param[in]  len  Number of bytes to checksum
+ *
+ * @return          Checksum of the specified memory area based on the
+ *                  given start value
+ */
+static inline uint16_t crc16_ccitt_aug_update(uint16_t crc, const unsigned char *buf, size_t len)
+{
+    /* Since CCITT-AUG and CCITT-FALSE only differ in the starting
+     * seed, we wrap around crc16_ccitt_false_update() for updating */
+    return crc16_ccitt_false_update(crc, buf, len);
+}
+
+/**
+ * @brief           Calculate CRC16-CCITT-AUG
+ * @details         polynom=0x1021
+ *                  seed=0x1d0f
+ *                  check=0xe5cc
```

Comment:
Is this what you intend the documentation to look like?

![Flat](https://user-images.githubusercontent.com/675644/187380064-b25d80ac-d99c-495b-ae43-80b258428c92.png)

In case you do not and my code-based proposal did not work for you (which would look like this)...

![Code](https://user-images.githubusercontent.com/675644/187381264-0d5e8b6c-2df9-4efb-99be-fd9b07a7ca51.png)

... a table could also be an option:

```suggestion
 *
 * Parameter | Value
 * --------: | :----
 * Polynom   | `0x1021`
 * Seed      | `0x1d0f`
 * Check     | `0xe5cc`
```

![Table](https://user-images.githubusercontent.com/675644/187382021-504961ec-e32a-4db2-9ca3-6e77113f8f0f.png)

Reply:
> [..] and my code-based proposal did not work for you (which would look like this)...

You proposal was great, I just forgot about it - should not have pushed last minute before heading home... 
I like the table idea - I picked it up and added some other values as well. Without going to much into detail: These parameter specify / identify one particular CRC algorithm, no more naming confusion! I initially omitted them since the `check` is already enough but the tables were just asking for more data. 😀

Thanks for providing your insights, I valued it a lot! :)